Automattic/wp-calypso
# Gutenberg: Disable opt-in on sites not supporting Gutenberg

#### Changes proposed in this Pull Request

We now check if the current site supports Gutenberg (determined by the changes to the Gutenberg opt-in API introduced in D30746-code) before enabling the opt-in flows.

This will prevent users from seeing the "Switch to the block editor" dialog if their sites are unable to load Gutenberg in WP.com (i.e.: conflicting active plugin or running a WP version lower than 5.0).

#### Testing instructions

* Make sure you don't have any active denied plugin for Gutenframe in your AT test site (see PCYsg-lOu-p2 for information about denied plugins).
* Go to `/block-editor/post` and select your AT test site.
* Add some content to the post.
* Save or publish the post.
* Switch to the classic editor from either the inline help or the more menu:
<img width="150" alt="Screen Shot 2019-07-24 at 12 05 56" src="https://user-images.githubusercontent.com/1233880/61762003-7c9c4480-ae0b-11e9-9db2-15f7cd79cd1e.png"><img width="100" alt="Screen Shot 2019-07-24 at 12 06 15" src="https://user-images.githubusercontent.com/1233880/61762008-7f973500-ae0b-11e9-964c-cbc0327bf598.png">

* Make sure a warning about the post containing blocks shows up:
<img width="300" alt="Screen Shot 2019-07-24 at 12 08 48" src="https://user-images.githubusercontent.com/1233880/61762080-cf75fc00-ae0b-11e9-8da0-8ff4a6738eb3.png">

* Click on the "Use the classic editor" button.
* Switch to the block editor from either the inline help or the sidebar:
<img width="120" alt="Screen Shot 2019-07-24 at 12 10 36" src="https://user-images.githubusercontent.com/1233880/61762171-1532c480-ae0c-11e9-880a-6d5c27669d0f.png"><img width="120" alt="Screen Shot 2019-07-24 at 12 10 26" src="https://user-images.githubusercontent.com/1233880/61762168-12d06a80-ae0c-11e9-8c12-c0f9d1758bbb.png">

* Activate a denied plugin.
* Go to `/posts` and select your AT test site.
* Open the post you created previously.
* Note how Calypso loads the classic editor.
* Make sure the warning about the post containing blocks doesn't show up.
* Make sure you cannot switch to the block editor from neither the inline help nor the sidebar.

Fixes #34856
